core/controllers/admin.py

```diff
@@ -127,6 +138,30 @@ def get(self):
     @require_super_admin
     def post(self):
         """Handles POST requests."""
+        def check_and_update_config_role(old_list, new_list, role):
```

Comment:
We should pull this out into a top-level function (outside this handler). It doesn't need to be an inner function.

Also, is the plan to delete it after the migration? If so it's worth saying that.